public_data/domain/consommation/progression/ConsommationProgressionService.py

```diff
@@ -0,0 +1,129 @@
+from django.db.models import Sum
+from django.db.models.query import QuerySet
+
+from public_data.domain.ClassCacher import ClassCacher
+from public_data.models import Cerema, Commune, Land
+
+from .ConsommationProgression import (
+    AnnualConsommation,
+    ConsommationProgressionAggregation,
+    ConsommationProgressionLand,
+)
+
+
+class ConsommationProgressionService:
+    def __init__(self, class_cacher: ClassCacher):
+        self.class_cacher = class_cacher
+
+    def get_by_communes_aggregation(
+        self,
+        communes: QuerySet[Commune],
+        start_date: int,
+        end_date: int,
+    ) -> ConsommationProgressionAggregation:
+        communes_key = "-".join(communes.values_list("insee", flat=True))
+        key = f"{start_date}-{end_date}-{communes_key}"
+
+        if self.class_cacher.exists(key):
+            return self.class_cacher.get(key)
+
+        surface_field = "surfcom23"
+
+        artif_fields = Cerema.get_art_field(
+            start=str(start_date),
+            end=str(end_date),
+        )
+        determinants = [
+            "hab",
+            "act",
+            "mix",
+            "rou",
+            "fer",
+            "inc",
+        ]
+        years = [str(i) for i in range(start_date, end_date + 1)]
+        determinants_fields = []
+
+        for year in years:
+            start = year[-2:]
+            end = str(int(year) + 1)[-2:]
+            for det in determinants:
+                determinants_fields.append(f"art{start}{det}{end}")
```

Comment:
```suggestion
        determinants_fields = [
            f"art{year[-2:]}{det}{str(int(year) + 1)[-2:]}"
            for year in years
            for det in determinants
        ]
```

Peut être légèrement plus lisible avec une compréhension de liste.